docs/develop/07_dApp/02_well-known-did-config.md

```diff
@@ -0,0 +1,109 @@
+---
+id: well-known-did-config
+title: Well Known DID Configuration
+---
+
+import TsJsSnippet from '@site/src/components/TsJsSnippet';
+import TsJsBlock from '@site/src/components/TsJsBlock';
+
+import DomainLinkageCtype from '!!raw-loader!@site/code_examples/dapp/src/dapp/domainLinkageCtype.ts';
+import DomainLinkageClaim from '!!raw-loader!@site/code_examples/dapp/src/dapp/domainLinkageClaim.ts';
+import SignPresentation from '!!raw-loader!@site/code_examples/dapp/src/dapp/signPresentation.ts';
+import AttestCredential from '!!raw-loader!@site/code_examples/dapp/src/dapp/attestCredential.ts';
+import FormatCredential from '!!raw-loader!@site/code_examples/dapp/src/dapp/formatCredential.ts';
+
+
+:::danger This is a working draft
+
+The KILT support of the *Well-Known DID Configuration* uses unpublished specifications and will change in the future.
+
+:::
+
+The *Well-Known DID Configuration* is implemented as a security measure when setting up the communication session between the dapp and extension.
+It ensures that the DID that the browser extension is communicating to is linked to the domain that is visited by the browser.
+This rule is currently enforced by the KILT DID Wallet reference implementation (Sporran Extension), but might be relaxed in the future.
+The implementation is based on the [*Well-Known DID Configuration*](https://identity.foundation/.well-known/resources/did-configuration/) specified by the Decentralized Identity Foundation.
+
+Once a communication session between a dapp and an extension is opened, the extension will query `<domain-name>/.well-known/did-configuration.json`.
+This JSON-file must contain a credential presentation that conforms to the [Domain Linkage CType][CType-Domain-Linkage].
+
+
+## Setup the Well-Known DID Configuration
+
+For the *Well-Known DID Configuration* you need to go through the following steps:
+
+0. Create a full DID
+    * You will need the `assertionMethodKey` a.k.a. `attestationKey` for signing the credential
+    * The `authenticationKey` is required for signing the transaction
+1. Create a claim
+2. Attest the claim
+3. Create a presentation
+4. Host the presentation on your website at `https://<your domain>/.well-known/did-configuration.json`
+
+### Create a DID
+
+Your dapp needs a DID to identify itself to the extension.
+If you don't have a DID yet, follow the [*create a full DID* guide](../01_sdk/02_cookbook/01_dids/02_full_did_creation.md).
+Make sure to create your DID with an `assertionMethodKey` so that you are able to issue attestations.
+
+### Making the claim
+
+After you got a DID, we can make a claim about that DID.
+The claim has to be based on the [Domain Linkage CType][CType-Domain-Linkage]:
+
+<TsJsBlock>
+  {DomainLinkageCtype}
+</TsJsBlock>
+
+The credential is built from the CType, claim contents, and your dapp's unique DID:
+
+<TsJsSnippet>
```

Comment:
Did you mean to use the snippet? I think it is not ideal that every snippet now has a return statement. Either we hide the return line, or we show the whole function including its signature.

Reply:
I wouldn't show the complete block since the `main` function takes some parameters which we don't necessarily want to show to the users. For me, the choice is either between the way things are now, or by removing the return line (which is possible). I prefer the latter, but I leave the final decision to you, as I don't have a strong opinion.

Reply:
[7907375](https://github.com/KILTprotocol/docs/pull/196/commits/7907375717ba778338261db1125faab29f31fb51)